### PR TITLE
feat: add debug rpc method to dump paymaster balances from pool

### LINKED ISSUE #573: [pool/rpc]: Expose paymaster balance tracking through a RPC endpoint
**Describe the feature**
Currently our paymaster tracking systems is a block boxes, we don't know what is going on while we are running. We should expose this information so that we can gain insight into our tracking and give feedback to users.

### PR DESCRIPTION
**Pull request by dancoombs**
Closes #573 

## Proposed Changes

  - Add new debug RPC method to dump and paymaster balances
  - Wire it all up

### CODE REVIEW

**Comment by dancoombs** [comment]
Likely going to remove the "check" endpoint from this, will just do it offline.

**Review by alex-miao — APPROVED** [review]
Looks good, only small thing is that we should add this method in `docs/architecture/rpc.md`

**Comment by dancoombs** [comment]
> Looks good, only small thing is that we should add this method in `docs/architecture/rpc.md`

Good call, added some docs and cleaned a few things up

**Comment by dancoombs** [comment]
Output looks like this:

```
curl 'localhost:3000' --header 'Content-Type: application/json' --data '{"method":"debug_bundler_dumpPaymasterBalances","params":["0x5FF137D4b0FDCD49DcA30c7CF57E578a026d2789"],"id":42,"jsonrpc":"2.0"}'
{"jsonrpc":"2.0","result":[{"address":"0xfb10850286e8f6b7b66bd60fb50b643afb8f13b1","pendingBalance":"0x316f486b5da791287","confirmedBalance":"0x316f486b5da791287"}],"id":42}% 
```

File: crates/pool/src/server/local.rs [hunk 1]
```diff
@@ -550,27 +608,15 @@ where
...
-                                        };
-                                        if let Err(e) = req.response.send(resp) {
-                                            tracing::error!("Failed to send response: {:?}", e);
-                                        }
-                                    });
-                                    continue;
-                                },
+                        ServerRequestKind::GetReputationStatus{ entry_point, address } => {
+                            match self.get_reputation_status(entry_point, address) {
+                                Ok(status) => Ok(ServerResponse::GetReputationStatus {  status }),
```
Comment (0xfourzerofour) [inline crates/pool/src/server/local.rs:619]: nit: fix spacing in `status`

File: crates/pool/src/server/local.rs [hunk 2]
```diff
@@ -447,6 +469,28 @@ where
         Ok(mempool.get_reputation_status(address))
     }
 
+    fn get_pool_and_spawn<F, Fut>(
+        &self,
+        entry_point: Address,
+        response: oneshot::Sender<Result<ServerResponse, PoolServerError>>,
+        f: F,
+    ) where
+        F: FnOnce(Arc<M>, oneshot::Sender<Result<ServerResponse, PoolServerError>>) -> Fut,
```
Comment (0xfourzerofour) [inline crates/pool/src/server/local.rs:478]: this cool

**Review by 0xfourzerofour — APPROVED** [review]
Good work on this

**Review by 0xfourzerofour — APPROVED** [review]

**Pull request merged**